nixos/modules/system/boot/systemd/dm-verity.nix

```diff
@@ -0,0 +1,65 @@
+{ config, lib, ... }:
+
+let
+  cfg = config.boot.initrd.systemd.dmVerity;
+in
+{
+  options = {
+    boot.initrd.systemd.dmVerity = {
+      enable = lib.mkEnableOption "dm-verity" // {
+        description = ''
+          Mount verity-protected block devices in the initrd.
+
+          Enabling this option allows to use `systemd-veritysetup` and
+          `systemd-veritysetup-generator` in the initrd.
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.enable -> config.boot.initrd.systemd.enable;
+        message = ''
+          'boot.initrd.systemd.dmVerity.enable' requires 'boot.initrd.systemd.enable' to be enabled.
+        '';
+      }
+    ];
+
+    boot.initrd = {
+      availableKernelModules = [
+        # For documentation, see https://docs.kernel.org/admin-guide/device-mapper/dm-init.html
+        "dm_mod"
+        # For documentation, see:
+        # - https://docs.kernel.org/admin-guide/device-mapper/verity.html
+        # - https://gitlab.com/cryptsetup/cryptsetup/-/wikis/DMVerity
+        "dm_verity"
+      ];
+
+      # dm-verity needs additional udev rules from LVM to work.
+      services.lvm.enable = true;
+
+      # The additional targets and store paths allow users to integrate verity-protected devices
+      # through the systemd tooling.
+      systemd = {
+        additionalUpstreamUnits = [
+          # https://github.com/systemd/systemd/blob/main/units/veritysetup-pre.target
```

Comment:
I think that's my doing, or at least my accidental prompting. When I review, I leave comments where I try to find out what the various constituent parts that are referenced land. Being able to show the research done -- as it were, notes on the margin of the page -- is a critical part of how I review that something is fit for purpose. 

* What does this service/tool do? 
* Where are its docs? 
* What other things can it do, other than what this particular PR lights up?

By commenting on the PR and linking the material, I hope to bring it into the conversation, so that others reviewing it might also bring new answers and new questions prompted by the source material.

I **don't** mean by commenting in the view that these URLs and reference material need to land in the source. Having them be clickable in the PR review is enough of a breadcrumb to follow, in my view. 

Though, unlike the view you express above, I don't find them particularly irksome.